p2p/discover/node.go

```diff
@@ -221,6 +221,13 @@ func (n NodeID) GoString() string {
 	return fmt.Sprintf("discover.HexID(\"%x\")", n[:])
 }
 
+// ShortID prints as 20 hex characters.
+type ShortID NodeID
+
+func (n ShortID) String() string {
+	return fmt.Sprintf("%x", n[:10])
+}
```

Comment:
I was thinking about adding an interface to the logger that would allow any type to have a `TerminalString` and `StructuredString` methods, which if set would be used to encode when printing in the logger.

This way we could add these methods to common.Hash, common.Address and the other major types without having to worry about redefining the String method (breaking some code), or having to write custom wrapping types.